test/functional/p2p_sendheaders.py

```diff
@@ -186,7 +183,7 @@ def check_last_inv_announcement(self, inv):
         inv should be a list of block hashes."""
 
         test_function = lambda: self.block_announced
-        wait_until(test_function, timeout=60, lock=mininode_lock)
+        self.wait_until(test_function)
```

Comment:
same

Reply:
This class extends P2PInterface not BitcoinTestFramework. mininode one already has a lock. As for timeout, it already has a default value of 60.